File: app/src/test/java/rx/TestInteractorSchedulers.java
package rx;

import com.lucasurbas.counter.app.di.rx.InteractorSchedulers;

import io.reactivex.Scheduler;
import io.reactivex.schedulers.Schedulers;

public class TestInteractorSchedulers implements InteractorSchedulers {

    @Override
    public Scheduler getBackgroundScheduler() {
        return Schedulers.trampoline();
    }

    @Override
    public Scheduler getMainThreadScheduler() {
        return Schedulers.trampoline();
    }

    @Override
    public Scheduler getSingleScheduler() {
        return Schedulers.trampoline();
    }
}

File: app/src/test/java/rx/TestInteractorSchedulers.kt
package rx

import com.lucasurbas.counter.app.di.rx.InteractorSchedulers

import io.reactivex.Scheduler
import io.reactivex.schedulers.Schedulers

class TestInteractorSchedulers : InteractorSchedulers {

    override val backgroundScheduler: Scheduler
        get() = Schedulers.trampoline()

    override val mainThreadScheduler: Scheduler
        get() = Schedulers.trampoline()

    override val singleScheduler: Scheduler
        get() = Schedulers.trampoline()
}
